src/hotspot/os/linux/os_perf_linux.cpp

```diff
@@ -383,11 +385,28 @@ static double get_cpu_load(int which_logical_cpu, CPUPerfCounters* counters, dou
     tdiff = udiff + kdiff;
   }
   *pkernelLoad = (kdiff / (double)tdiff);
+  user_load = (udiff / (double)tdiff);
+
+  if (OSContainer::is_containerized()) {
+    double scale = 1.0f;
+    int quota  = OSContainer::cpu_quota();
+    int period = OSContainer::cpu_period();
+    int share  = OSContainer::cpu_shares();
+    int processor_count = os::processor_count();
+    if (quota > -1 && period > 0) {
+      scale = (double)processor_count / ((double)quota / (double)period);
+    } else if (share > -1) {
+      scale = (double)processor_count / ((double)share / (double)PER_CPU_SHARES);
+    } else {
+      scale = (double)processor_count / (double)os::active_processor_count();
+    }
+    *pkernelLoad *= scale;
+    user_load *= scale;
```

Comment:
If I read this correctly the cpu load would only be scaled to the number of configured container limits with no bearing of the actual cpu usage of processes within the container. For example: Let process A be running in container A' with X/2 cpus and process B running in container B' with X/2 cpus. X are the total number of cpus available on a host. Then A is using 100% cpu (i.e. `1.0`) and B is being idle (i.e. close to `0.0`). With this patch I'd get a CPU load report of about `1.0` for each container, A' and B'. Am I missing something?

On the JDK side we have `Metrics.getCpuUsage()` and APIs which isn't present in HotSpot code. I believe we'd have to have it in OSContainer API in order to implement this correctly. Thoughts?

Reply:
My proposal shows correctly both `jvmUser` and `jvmSystem` in your case, however `machineTotal` is incorrect (100%). I need to tweak it.

> On the JDK side we have `Metrics.getCpuUsage()` API which isn't present in HotSpot code. I believe we'd have to have a similar API in OSContainer in order to implement this correctly. Thoughts?

It is better to do so, but I thought it is much simpler if we can handle CPU quotas correctly.
I understand it is better if it is same logic between JDK and HotSpot, but I thought we can work for it in future RFE. Isn't it?

Reply:
Further investigation, I think need to read cpuacct.usage in cgroups v1, or cpu.stat in cgroups v2 when `get_cpu_load()` is called with `CPU_LOAD_GLOBAL` - @jerboaa did you point it out?

Of course I can add it to this PR, but if we work for it, I think we should consolidate implementation - but it may not happen until RDP 1 (and it might be big change).

Hence I will withdraw this PR and will close JBS, is it ok?

Reply:
@YaSuenag Yes, that's what `Metrics.getCpuUsage()` does under the hood. I'd suggest to first add the `get_cpu_usage()` API to `OSContainer` and then once it's there use it here. Or maybe consolidate things and you'll get away with using the Java API. Either way I'm fine with waiting/withdrawing.

Reply:
Thanks @jerboaa !
I will withdraw this PR and will close JBS. I hope we will consolidate logics in near future...